### PR TITLE
*: fix table namespace classifier.

### PR DESCRIPTION
**Pull request by disksing**
1. _GetAllNamespaces_ should contain _DefaultNamespace_, or the stores in default namespace will not be balanced.
2. _ IsPureTableID_ need decode the key before check if it is table prefix.

### CODE REVIEW

File: table/codec.go [hunk 1]
```diff
@@ -84,7 +84,11 @@ func decodeCmpUintToInt(u uint64) int64 {
 
 // IsPureTableID return true iff b is consist of tablePrefix and 8-byte tableID
```
Comment (nolouch) [inline table/codec.go]: s/iff/if

File: table/codec.go [hunk 2]
```diff
@@ -82,9 +82,13 @@ func decodeCmpUintToInt(u uint64) int64 {
 	return int64(u ^ signMask)
 }
 
-// IsPureTableID return true iff b is consist of tablePrefix and 8-byte tableID
+// IsPureTableID returns true if b is consist of tablePrefix and 8-byte tableID
 func IsPureTableID(b []byte) bool {
```
Comment (siddontang) [inline table/codec.go]: please add test

File: table/namespace_classifier.go [hunk 3]
```diff
@@ -123,6 +123,7 @@ func (c *tableNamespaceClassifier) GetAllNamespaces() []string {
 	for name := range c.nsInfo.namespaces {
 		nsList = append(nsList, name)
 	}
+	nsList = append(nsList, namespace.DefaultNamespace)
```
Comment (siddontang) [inline table/namespace_classifier.go]: need to check duplicated default here?

Comment (disksing) [reply table/namespace_classifier.go]: Maybe forbid creating namespace using `global` instead.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**